index.js

```diff
@@ -1,76 +1,106 @@
-                destDir: 'app'
-            });
-
-            var tests = this.treeGenerator(path.join(this.project.root,  'tests'));
-
-            var blanketOptions = new Funnel(tests, {
-                files: ['blanket-options.js'],
-                srcDir: '/',
-                destDir: '/assets'
-            });
-
-            var lib = this.treeGenerator(path.join(__dirname, 'lib'));
-
-            var blanketLoader = new Funnel(lib, {
-                files: ['blanket-loader.js'],
-                srcDir: '/',
-                destDir: '/assets'
-            });
-
-            var start = new Funnel(lib, {
-                files: ['start.js'],
-                srcDir: '/',
-                destDir: '/'
-            });
-
-            var testLoaderTree = this.concatFiles(mergeTrees([treeTestLoader, start]), {
-                inputFiles: ['**/*.js'],
-                outputFile: '/assets/test-loader.js'
-            });
-
-            return mergeTrees([tree, blanketOptions, blanketLoader, testLoaderTree], {
-                overwrite: true
-            });
-        }
-        else {
-            return tree;
-        }
+  name: 'Ember CLI Blanket',
+
+  validEnv: function() {
+    return this.app.env !== 'production';
+  },
+
+  blueprintsPath: function() {
+    return path.join(__dirname, 'blueprints');
+  },
+
+  included: function included(app) {
+    this._super.included(app);
+
+    if (app.tests) {
+      var fileAssets = [
+        path.join(app.bowerDirectory, 'blanket', 'dist', 'qunit', 'blanket.js'),
+      ];
+
+      fileAssets.forEach(function(file) {
+        app.import(file, {
+          type: 'test'
+        });
+      });
+    }
+  },
+
+  middleware: function(app, options) {
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({
+      extended: true
+    }));
+    app.use(coverageMiddleware(options));
+    app.use(logErrors);
+  },
+  testemMiddleware: function(app) {
+    this.middleware(app, { root: this.project.root });
+  },
+  serverMiddleware: function(options) {
+    this.app = options.app;
+    if(!this.validEnv()) { return; }
+    this.middleware(options.app, { root: this.project.root });
+  },
+  postprocessTree: function(type, tree) {
+    this._requireBuildPackages();
+
+    if (type === 'all' && this.app.tests) {
+      var treeTestLoader = new Funnel(tree, {
+        files: ['test-loader.js'],
+        srcDir: 'assets',
+        destDir: 'app'
+      });
+
+      var tests = this.treeGenerator(path.join(this.project.root, 'tests'));
+
+      var blanketOptions = new Funnel(tests, {
+        files: ['blanket-options.js'],
+        srcDir: '/',
+        destDir: '/assets'
+      });
+
+      var vendor = this.treeGenerator(path.join(__dirname, 'vendor'));
+
+      var blanketLoader = new Funnel(vendor, {
+        files: ['blanket-reporter.js', 'blanket-require.js'],
+        srcDir: '/',
+        destDir: '/'
+      });
+
+      blanketLoader = this.concatFiles(blanketLoader, {
```

Comment:
we're concatting multiple needed files into one - the rename on blanket-loader.js to blanket-require.js was to reduce confusion by Chrome on sourcemaps - source file and concatted file were the same name